pkg/server/test/list_objects.go

```diff
@@ -126,6 +99,31 @@ func TestListObjectsRespectsMaxResults(t *testing.T, ds storage.OpenFGADatastore
 			allResults:             []string{"org:1", "org:2", "org:3"},
 			useCheckCache:          false,
 		},
+		{
+			name:   "max_results_with_model_that_uses_exclusion_and_one_object_is_a_false_candidate",
```

Comment:
This test fails on `main` 😅 
<img width="813" alt="image" src="https://github.com/openfga/openfga/assets/5374887/1e856d5a-ce38-45b2-9eee-6f6b10b30cb0">

In `main` we are passing `maxResults: 2` to ConnectedObjects, which gives us 2 objects (org:2 and org:1). But `org:1` has to be analyzed via Check, which returns false, so in reality we should have kept on looking for candidates.